### PR TITLE
fix(nats-connector): change stream into optional string, add replace stream name logic

### PR DESCRIPTION
**Pull request by yufansong**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?
1. change `stream` filed into optional, in source, we require nats jetstream, the stream is required. But in sink, we support nats, then stream can be empty. So we change `stream` into optional and add check logic in source code.
2. Nats require no `.` in consumer name, we add replace logic in code to avoid this situation happend.

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [ ] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [ ] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

If this PR includes changes that directly affect users or other significant modifications relevant to the community, kindly draft a release note to provide a concise summary of these changes. Please prioritize highlighting the impact these changes will have on users.

</details>

### CODE REVIEW

File: src/connector/src/common.rs [hunk 1]
```diff
@@ -512,7 +512,10 @@ impl NatsCommon {
     > {
         let context = self.build_context().await?;
         let stream = self.build_or_get_stream(context.clone()).await?;
-        let subject_name = self.subject.replace(',', "-");
+        let subject_name = self
+            .subject
+            .replace(',', "-")
+            .replace(['.', '>', '*', ' ', '\t'], "_");
```
[the remark this answers is not in the document]
Comment (yufansong) [reply src/connector/src/common.rs:518]: 1. After communicating with customer, in their setting, they already set their subject with `.`, so maybe need to do the replace, or they need to reset all their production environment.
2. This `subject_name` will only used when create `consumer name`. We will keep the `self.subject` and pass it into nats as subject name.

File: src/connector/src/common.rs [hunk 2]
```diff
@@ -417,10 +417,10 @@ pub struct UpsertMessage<'a> {
 pub struct NatsCommon {
     #[serde(rename = "server_url")]
     pub server_url: String,
-    #[serde(rename = "stream")]
-    pub stream: String,
     #[serde(rename = "subject")]
     pub subject: String,
+    #[serde(rename = "stream")]
+    pub stream: Option<String>,
```
Comment (tabVersion) [inline src/connector/src/common.rs:423]: why option

**Review by tabVersion — APPROVED** [review]
rest LGTM

File: src/connector/src/common.rs (hunk 2 above)
Comment (yufansong) [reply src/connector/src/common.rs]: Discussed offline, move stream from common into properties.

**Pull request merged**